### PR TITLE
feat(md): formats md before committing

### LINKED ISSUE #850: feat(docs): addition of markdownlint
**Is your feature request related to a problem? Please describe.**
Inconsistent formatting and linting of docs, throughout the project.

**Motivation**
While working for the PR #842, @ematipico  pointed out some linting issues :+1: .
Hence linting must be prioritised, inspired by the Webpack.js.org project, which follows the same standards to [lint the markdown](https://github.com/webpack/webpack.js.org/blob/d99dc8ca1f2e60208efe893c03c06280ed24e5f1/package.json#L42).

**Describe the solution you'd like**
Addition of a markdown linting pipeline.

**Describe alternatives you've considered**
Can't say.

Let's discuss :+1:

### PR DESCRIPTION
**Pull request by pranshuchittora**
Closes #850 

Linting of markdown, following webpack.js.org standards
### Why tho?
- Consistent formatting.
- Easy to migrate docs to webpack.js.org.
- It's good to follow some standards.

**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
Nopes

**Summary**

**Does this PR introduce a breaking change?**
Most probably not.
Nopes

---
### Milestones :medal_sports: 

- [ ] Addition of markdownlint

- [ ] Setup base config

- [ ] Integrate with `husky`

- [ ] Fix all linting issue in all the docs :100:

### CODE REVIEW

**Comment by anshumanv** [comment]
I don't think we have so much md like [docs](https://webpack.js.org) site that we need a separate linter and introduce more deps in the package, it can be easily maintained without it and the markdown here(most of it) is gonna be shifted entirely to docs site so this will be even less used.

(IMO)

**Comment by ematipico** [comment]
I think what's best is to add prettier to the lint-staged hook command and tell it to format also the .MD files. Having a linter is too much I think :)

**Comment by pranshuchittora** [comment]
@ematipico  Sounds great, will be working on that :+1:

**Comment by ematipico** [comment]
Could you please rebase? I think it's good to go!

File: package.json [hunk 1]
```diff
@@ -124,6 +128,7 @@
     "import-local": "^2.0.0",
     "interpret": "^1.1.0",
     "loader-utils": "^1.1.0",
+    "prettier": "^1.17.0",
```
Comment (rajivshah3) [inline package.json]: @pranshuchittora was there a reason `prettier` needed to be added to `dependencies` instead of `devDependencies`?

Comment (andersk) [reply package.json]: Fix submitted as #968.

**Pull request merged**